CRM/Contribute/BAO/ManagePremiums.php

```diff
@@ -127,16 +118,16 @@ public static function del($productID) {
     $premiumsProduct = new CRM_Contribute_DAO_PremiumsProduct();
     $premiumsProduct->product_id = $productID;
     if ($premiumsProduct->find(TRUE)) {
-      $session = CRM_Core_Session::singleton();
-      $message .= ts('This Premium is being linked to <a href=\'%1\'>Online Contribution page</a>. Please remove it in order to delete this Premium.', array(1 => CRM_Utils_System::url('civicrm/admin/contribute', 'reset=1')), ts('Deletion Error'), 'error');
-      CRM_Core_Session::setStatus($message);
-      return CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/admin/contribute/managePremiums', 'reset=1&action=browse'));
+      $message = ts('This Premium is being linked to <a href=\'%1\'>Online Contribution page</a>. Please remove it in order to delete this Premium.', array(1 => CRM_Utils_System::url('civicrm/admin/contribute', 'reset=1')));
+      CRM_Core_Session::setStatus($message, ts('Deletion Error'), 'error');
+      CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/admin/contribute/managePremiums', 'reset=1&action=browse'));
```

Comment:
Can we throw exception with a message here and catch at form level to redirect to 'civicrm/admin/contribute/managePremiums'? 

I believe if i try to delete Premiums through api it will actually return success but on screen it will pop with a message.

Reply:
Good idea. I've done that.